app/controllers/api/v1/search_controller.rb

```diff
@@ -1,31 +1,22 @@
 class Api::V1::SearchController < ApplicationController
   def find_merchant
-    if params[:name] && params[:name].present?
-      merchant =
-      Merchant.where('name Ilike ?', "%#{params[:name]}%")
-      .order('LOWER(name)')
-      .first
-
-      if merchant
-        render json: MerchantSerializer.new(merchant)
-      else
-        blank_merchant = Merchant.create
-        render json: MerchantSerializer.new(blank_merchant), status: 400
-      end
+    if params[:name]&.present?
+      merchant = Merchant.find_match_by_name(params[:name]).first
+      render json: MerchantSerializer.new(merchant) if merchant.present?
+      render json: MerchantSerializer.new(Merchant.create), status: :bad_request if merchant.nil?
+      # I would have just created a no found error for this
+      # render json: { error: 'Merchant not found' }.to_json, status: :not_found if merchant.nil?
```

Comment:
Used guard clauses to prevent nested if statements

Reply:
This is hella clean! Great point here 😄